django/db/models/query.py

```diff
@@ -288,7 +288,13 @@ def aggregate(self, *args, **kwargs):
         if self.query.distinct_fields:
             raise NotImplementedError("aggregate() + distinct(fields) not implemented.")
         for arg in args:
-            if not hasattr(arg, 'default_alias'):
+            # The default_alias property may raise a TypeError, so we use
+            # a try/except construct rather than hasattr in order to remain
+            # consistent between PY2 and PY3 (hasattr would swallow
+            # the TypeError on PY2).
+            try:
+                arg.default_alias
+            except (AttributeError, TypeError):
                 raise TypeError("Complex aggregates require an alias")
```

Comment:
I'm not too sure, what would it bring? That the `TypeError("Complex aggregates require an alias")` started life as another exception is more an implementation detail imo.